File: pyround1/1502.判断能否形成等差数列.py
#
# @lc app=leetcode.cn id=1502 lang=python3
#
# [1502] 判断能否形成等差数列
#

# @lc code=start
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        ls = sorted(arr)
        diff = ls[1] - ls[0]
        for i in range(1, len(arr)-1):
            if ls[i+1] - ls[i] != diff:
                return False
        return True
# @lc code=end

File: 1502.判断能否形成等差数列.cpp
/*
 * @lc app=leetcode.cn id=1502 lang=cpp
 *
 * [1502] 判断能否形成等差数列
 */

// @lc code=start
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        if(arr.size() < 2)
            return true;
        sort(arr.begin(), arr.end());
        int fast = 1, slow = 0;
        int divide = arr[1] - arr[0];
        while(fast < arr.size() and arr[fast] - arr[slow] == divide){
            fast++;
            slow++;
        }
        if(fast == arr.size())
            return true;
        return false;
    }
};
// @lc code=end

